lib/index.js

```diff
@@ -19,13 +19,20 @@ module.exports = {
 
     validCodegen.forEach((codegen) => {
       main = codegen.main;
-      if (typeof main.getOptions !== 'function') {
-        return callback('Codegen~getOptions: getOptions is not a function');
-      }
-      if (!main.getOptions) {
-        return callback('Codegen~convert: Could not find condegen corresponding to provided language, variant pair');
+      try {
+        if (typeof main.getOptions !== 'function') {
+          throw TypeError('Codegen~getOptions: getOptions is not a function');
+        }
+        if (!main.getOptions) {
+          throw ReferenceError('Codegen~convert: Could not find condegen corresponding to provided language, variant pair');
+        }
+      } catch(e) {
+        if (e instanceof TypeError || e instanceof ReferenceError) {
+          console.log(e.message)
```

Comment:
Adding console.log for an error doesn't make sense here, if the `return` will be called anyway with `callback(null, main.getOptions());` 

The whole point of the check is to not call `getOptions()` if it is not a function.